tracer/src/Datadog.Trace/Debugger/Symbols/AssemblyFilter.cs

```diff
@@ -16,16 +16,31 @@ internal class AssemblyFilter
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AssemblyFilter));
 
-        internal static bool ShouldSkipAssembly(Assembly assembly, HashSet<string>? includeList = null)
+        internal static bool ShouldSkipAssembly(Assembly assembly, HashSet<string>? excludeList = null, HashSet<string>? includeList = null)
         {
             var assemblyName = assembly.GetName().Name;
-            return string.IsNullOrWhiteSpace(assemblyName) ||
-                   assembly.IsDynamic ||
-                   assembly.ManifestModule.IsResource() ||
-                   string.IsNullOrWhiteSpace(assembly.Location) ||
-                   IsDatadogAssembly(assemblyName) ||
-                   IsThirdPartyCode(assemblyName) ||
-                   (includeList != null && !IsInIncludeList(assemblyName, includeList));
+
+            var shouldSkip = string.IsNullOrWhiteSpace(assemblyName) ||
+                             assembly.IsDynamic ||
+                             assembly.ManifestModule.IsResource() ||
+                             string.IsNullOrWhiteSpace(assembly.Location);
+
+            if (shouldSkip)
+            {
+                return true;
+            }
+
+            if (IsInExcludeList(assemblyName!, excludeList))
+            {
+                return false;
+            }
```

Comment:
Is this the required behavior? Assemblies in `excludeList` should **not** be skipped?

Reply:
yes, because it means that those assemblies are exclude from the 3rd party detection list i.e. do not skip them